lib/rules/prefer-regex-literals.js

```diff
@@ -170,7 +306,83 @@ module.exports = {
                             context.report({ node, messageId: "unexpectedRedundantRegExp" });
                         }
                     } else if (hasOnlyStaticStringArguments(node)) {
-                        context.report({ node, messageId: "unexpectedRegExp" });
+                        let regexContent = getStringValue(node.arguments[0]);
+                        let noFix = false;
+                        let flags;
+
+                        if (node.arguments[1]) {
+                            flags = getStringValue(node.arguments[1]);
+                        }
+
+                        const regexppEcmaVersion = getRegexppEcmaVersion(context.parserOptions.ecmaVersion);
+                        const RegExpValidatorInstance = new RegExpValidator({ ecmaVersion: regexppEcmaVersion });
+
+                        try {
+                            RegExpValidatorInstance.validatePattern(regexContent, 0, regexContent.length, flags ? flags.includes("u") : false);
+                            if (flags) {
+                                RegExpValidatorInstance.validateFlags(flags);
+                            }
+                        } catch {
+                            noFix = true;
+                        }
+
+                        const tokenBefore = sourceCode.getTokenBefore(node);
+
+                        if (tokenBefore && !validPrecedingTokens.includes(tokenBefore.value)) {
+                            noFix = true;
+                        }
+
+                        if (!/^[-a-zA-Z0-9\\[\](){}\s!@#$%^&*+^_=/~`.><?,'"|:;]*$/u.test(regexContent)) {
+                            noFix = true;
+                        }
+
+                        if (regexContent && !noFix) {
+                            let charIncrease = 0;
+
+                            const ast = new RegExpParser({ ecmaVersion: regexppEcmaVersion }).parsePattern(regexContent, 0, regexContent.length, flags ? flags.includes("u") : false);
+
+                            visitRegExpAST(ast, {
+                                onCharacterEnter(characterNode) {
+                                    let changeCharIncrease = 0;
+                                    let stringCodePointValue = String.fromCodePoint(characterNode.value);
+                                    const escaped = resolveEscapes(stringCodePointValue);
```

Comment:
This would be incorrectly transformed:

```js
new RegExp("\\u000A\\u000A");
```

After the suggestion is applied:

```js
/\n\u00\n/;
```

Instead of `characterNode.value`, we should check `characterNode.raw`, and convert it only if it's a problematic character (for example, `<LF>`), or an escaped problematic character (for example, `\<LF>`).

```js
new RegExp("\u000A"); // -> /\n/

new RegExp("\\u000A"); // -> /\u000a/

new RegExp("\\\u000A"); // -> /\n/
```